### PR TITLE
fix function error in keywords of block.number/hash/timestamp

### PR DESCRIPTION
**Pull request by KamiD**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/watcher/querier.go [hunk 1]
```diff
@@ -80,6 +80,25 @@ func (q Querier) GetBlockByHash(hash common.Hash, fullTx bool) (*EthBlock, error
 	return &block, nil
 }
 
+func (q Querier) GetBlockHashByNumber(number uint64) ([]byte, error) {
```
Comment (xiangjianmeng) [inline x/evm/watcher/querier.go]: returning (common.Hash, error) maybe better

File: x/evm/watcher/querier.go [hunk 2]
```diff
@@ -80,6 +80,25 @@ func (q Querier) GetBlockByHash(hash common.Hash, fullTx bool) (*EthBlock, error
...
+	if !q.enabled() {
+		return nil, errors.New(MsgFunctionDisable)
+	}
+	var height = number
```
Comment (xiangjianmeng) [inline x/evm/watcher/querier.go:87]: why not use number directly

File: app/rpc/namespaces/eth/simulation/impl.go [hunk 3]
```diff
@@ -201,10 +203,57 @@ func (i InternalDba) NewStore(parent store.KVStore, Prefix []byte) evmtypes.Stor
...
 	return nil
 }
 
+type HeightHashStore struct {
+	q *watcher.Querier
+}
+
+func (s HeightHashStore) Set(key, value []byte) {
+	//just ignore all set opt
+	return
```
Comment (xiangjianmeng) [inline app/rpc/namespaces/eth/simulation/impl.go]: delete return maybe better

File: x/evm/watcher/querier.go (hunk 2 above)
Comment (KamiD) [reply x/evm/watcher/querier.go:87]: because there has another condition to change the variable of height
```go
height, err = q.GetLatestBlockNumber()
```

**Review by xiangjianmeng — APPROVED** [review]

**Review by ilovers — APPROVED** [review]

**Review by summerpro — APPROVED** [review]

**Pull request merged**